.update_api.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import print_function
+import itertools
+
+if __name__ != "__main__":
+    import sys
+    sys.exit("This should never be imported as a library")
+
+# Replace module Ref, drop module Legacy and remove Stdext from aPI.ml
+with open('lib/aPI.ml', 'r') as api_f:
+    api_lines = api_f.readlines()
+
+with open('.ref_replacement', 'r') as ref_f:
+    ref_replacement = ref_f.readlines()[:-1]  # 'end\n' is already in the tail
+
+
+api_head = list(
+    itertools.takewhile(lambda line: "module Ref" not in line,
```

Comment:
Maybe this will be good enough in practice, but it does seem rather brittle: for example the search for `"module Ref"` could match on a comment or string or `module RefFooBar`, or could fail to match on `module  Ref` (which has two spaces instead of one).

Similar issues for the functions that examine lines for `"module Legacy"` and `"Stdext"`.

Reply:
It is, but I think it is the least evil in this case. The aPI file is autogenerated and the autogeneration is not going to change anytime soon, any mess up with those name and anyways the module will fail to compile.

I am working on breaking the cyclic dependency between xapi and xen-api-client, after which we can completely get rid of this part and rely directly on `xapi-client`

Reply:
Right, this will be safe enough.